pkg/code-generator/codegen/templates/resource_template.go

```diff
@@ -87,8 +87,66 @@ func (r *{{ .Name }}) SetMetadata(meta *core.Metadata) {
 {{- if $.HasStatus }}
 
 func (r *{{ .Name }}) SetStatus(status *core.Status) {
-	r.Status = status
+	r.UpsertReporterStatus(status)
 }
+
+func (r *{{ .Name }}) SetReporterStatus(status *core.ReporterStatus) {
+	r.StatusOneof = &{{ .Name }}_ReporterStatus{ReporterStatus: status}
+}
+
+// UpsertReporterStatus inserts the specified status into the ReporterStatus.Statuses map for the
+// current namespace (as specified by POD_NAMESPACE env var).  If the resource does not yet have
+// a ReporterStatus, one will be created.
+// Note: POD_NAMESPACE environment variable must be set for this function to behave as expected.
+func (r *{{ .Name }}) UpsertReporterStatus(status *core.Status) {
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	if podNamespace != "" {
```

Comment:
Should we be doing something here if `POD_NAMESPACE` env var is not set? Some sort of error reporting? This can silently fail if os.getenv(POD_NAMESPACE) fails to get the pod namespace..

Reply:
Good idea, this wouldn't be a fun issue to track down. Going to do the same for `GetNamespacedStatus`